go/multipooler/manager/rpc_consensus.go

```diff
@@ -0,0 +1,284 @@
+	}
+
+	pm.mu.Lock()
+	pm.consensusTerm.VotedFor = candidateID
+	termToSave := pm.consensusTerm
+	pm.mu.Unlock()
+
+	if err := SetTerm(pm.config.PoolerDir, termToSave); err != nil {
+		return nil, fmt.Errorf("failed to persist vote: %w", err)
+	}
+
+	response.Accepted = true
+	return response, nil
+}
+
+// ConsensusStatus returns the current status of this node for consensus
+func (pm *MultiPoolerManager) ConsensusStatus(ctx context.Context, req *consensusdatapb.StatusRequest) (*consensusdatapb.StatusResponse, error) {
+	// Load consensus term from disk if not already loaded
+	pm.mu.Lock()
+	if pm.consensusTerm == nil {
+		term, err := GetTerm(pm.config.PoolerDir)
+		if err != nil {
+			pm.mu.Unlock()
+			return nil, fmt.Errorf("failed to load consensus state: %w", err)
+		}
+		pm.consensusTerm = term
+	}
+
+	// Get local voting term from file
+	localTerm := pm.consensusTerm.GetCurrentTerm()
+	pm.mu.Unlock()
+
+	// Get last successful leader term from Postgres (replicated data)
+	var leaderTerm int64
+	isHealthy := false
+
+	if pm.db != nil {
+		err := pm.db.QueryRowContext(ctx, `
+			SELECT COALESCE(MAX(leader_term), 0)
+			FROM multigres.heartbeat
+			WHERE shard_id = $1
+		`, []byte(req.ShardId)).Scan(&leaderTerm)
+
+		if err != nil {
+			pm.logger.Warn("Failed to query postgres leader term", "error", err)
+			leaderTerm = 0
+		} else {
+			isHealthy = true
+		}
+	}
+
+	// Get current WAL position
+	walPosition := &consensusdatapb.WALPosition{}
+	if isHealthy {
+		var err error
+		walPosition, err = pm.GetCurrentWALPosition(ctx)
+		if err != nil {
+			pm.logger.Warn("Failed to get WAL position", "error", err)
+		}
+	}
+
+	// Determine role (primary/replica)
+	role := "replica"
+	if isHealthy {
+		var inRecovery bool
+		err := pm.db.QueryRowContext(ctx, "SELECT pg_is_in_recovery()").Scan(&inRecovery)
+		if err == nil && !inRecovery {
+			role = "primary"
+		}
+	}
+
+	return &consensusdatapb.StatusResponse{
+		PoolerId:    pm.serviceID.GetName(),
+		CurrentTerm: localTerm,
+		LeaderTerm:  leaderTerm,
+		WalPosition: walPosition,
+		IsHealthy:   isHealthy,
+		IsEligible:  true, // TODO: implement eligibility logic based on policy
+		Cell:        pm.serviceID.GetCell(),
+		Role:        role,
+	}, nil
+}
+
+// GetLeadershipView returns leadership information from the heartbeat table
+func (pm *MultiPoolerManager) GetLeadershipView(ctx context.Context, req *consensusdatapb.LeadershipViewRequest) (*consensusdatapb.LeadershipViewResponse, error) {
+	if pm.replTracker == nil {
+		return nil, fmt.Errorf("replication tracker not initialized")
+	}
+
+	// Use the heartbeat reader to get leadership view
+	reader := pm.replTracker.HeartbeatReader()
+	view, err := reader.GetLeadershipView()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get leadership view: %w", err)
+	}
+
+	return &consensusdatapb.LeadershipViewResponse{
+		LeaderId:          view.LeaderID,
+		LeaderTerm:        view.LeaderTerm,
+		LeaderWalPosition: view.LeaderWALPosition,
```

Comment:
Thinking we should not expose this WAL here. This will be lagged and it might not be the actual LeaderWalPosition.